src/main/java/cloud/eppo/api/Configuration.java

```diff
@@ -230,4 +241,45 @@ public Configuration build() {
           flags, banditReferences, bandits, isConfigObfuscated, flagJson, banditParamsJson);
     }
   }
+
+  /**
+   * Contains the base data required to hydrate a `Configuration` object
+   *
+   * <p>Using `Configuration.writeToSteam` and `Configuration.readFromStream` allows us to avoid
+   * implementing JSON serialization for the `Configuration` class or making all the DTOs `implement
+   * Serializable`
+   */
+  private static class ConfigurationPacket implements Serializable {
+    final byte[] flagConfigBytes;
+    final byte[] banditParamsBytes;
+    final boolean isConfigObfuscated;
+
+    ConfigurationPacket(
+        byte[] flagConfigBytes, byte[] banditParamsBytes, boolean isConfigObfuscated) {
+      this.flagConfigBytes = flagConfigBytes;
+      this.banditParamsBytes = banditParamsBytes;
+      this.isConfigObfuscated = isConfigObfuscated;
+    }
+  }
+
+  private ConfigurationPacket getConfigurationPacket() {
+    return new ConfigurationPacket(flagConfigJson, banditParamsJson, isConfigObfuscated);
+  }
+
+  public void writeToStream(OutputStream out) throws IOException {
+    ObjectOutputStream oos = new ObjectOutputStream(out);
```

Comment:
❗ didn't know about this

However is this the planned use for caching in the filesystem? I think for that we should just save off the original received JSON

Reply:
refactored everything to save the JSON bytes directly to disk. Requires "grafting" the `forServer` field onto the `FlagConfigResponse` json which in turn, requires a deserialization+reserialization before the JSON string is written to disk. This, of course, will be unnecessary when `forServer` is added to the FCR